applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -0,0 +1,366 @@
+#include "cloud/cloud_wrapper.h"
+#include <zephyr.h>
+#include <net/nrf_cloud.h>
+#include <net/mqtt.h>
+
+#include "cJSON.h"
+#include "json_helpers.h"
+#include "json_protocol_names.h"
+
+#define MODULE nrf_cloud_integration
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_CLOUD_INTEGRATION_LOG_LEVEL);
+
+#define NRF_CLOUD_SERVICE_INFO "{"								\
+					"\"state\":{"						\
+						"\"reported\":{"				\
+							"\"device\":{"				\
+								"\"serviceInfo\":{"		\
+									"\"ui\":["		\
+										"\"GPS\","	\
+										"\"HUMID\","	\
+										"\"RSRP\","	\
+										"\"BUTTON\","	\
+										"\"TEMP\""	\
+									"],"			\
+									"\"fota_v2\":["		\
+										"\"APP\","	\
+										"\"MODEM\","	\
+										"\"BOOT\""	\
+									"]"			\
+								"}"				\
+							"}"					\
+						"}"						\
+					"}"							\
+				"}"
+
+#define REQUEST_DEVICE_STATE_STRING ""
+
+static cloud_wrap_evt_handler_t wrapper_evt_handler;
+
+static void cloud_wrapper_notify_event(const struct cloud_wrap_event *evt)
+{
+	if ((wrapper_evt_handler != NULL) && (evt != NULL)) {
```

Comment:
Can be simplified to `if (wrapper_evt_handler  && evt)`

Reply:
Lets just keep it like this and do a commit later which changes this for all integration layers.

Reply:
ack